### PR TITLE
Topkg

### PR DESCRIPTION
**Pull request by hannesm**
on top of #19, let's see whether CI likes it

### CODE REVIEW

**Comment by hannesm** [comment]
I fail to get the tests to run properly under travis.

**Comment by yomimono** [comment]
I don't think they ran under Travis before this PR: https://travis-ci.org/mirage/ocaml-tuntap/jobs/165253097 just installs and removes AFAICT.

**Comment by hannesm** [comment]
@yomimono thx, I disabled tests in e55fb4b.  This is ready to merge (thx @gjaldon for the majority of the effort) :)

File: _tags [hunk 1]
```diff
@@ -1,70 +1,8 @@
...
-<test/*.ml{,i,y}>: pkg_ipaddr
-<test/*.ml{,i,y}>: use_tuntap
-# OASIS_STOP
-<lib>: include
 true: annot, bin_annot, debug
+true: package(ipaddr)
+
+<lib>: include
+
+<lib/tuntap.{cma,cmxa}>: use_tuntap
```
Comment (avsm) [inline _tags]: is this `use_tuntap` still needed here?

Comment (hannesm) [inline _tags]: yes, otherwise we've link errors from missing symbols (originating from `tuntap_stubs`, see @gjaldon comment in #19)

Comment (avsm) [inline _tags]: ah yes, makes sense. thanks!

**Comment by yomimono** [comment]
I think we still have link errors from missing symbols in the released 1.4.0:

```
ocamlfind ocamlopt -g -linkpkg -g -package tcpip.udp -package tcpip.tcp -package tcpip.stack-direct -package tcpip.ipv4 -package tcpip.icmpv4 -package tcpip.ethif -package tcpip.arpv4 -package tcpip -package nocrypto.lwt -package nocrypto -package mirage-unix -package mirage-types-lwt -package mirage-types -package mirage-runtime -package mirage-random -package mirage-net-unix -package mirage-logs -package mirage-http -package mirage-clock-unix -package lwt -package functoria-runtime -package conduit.mirage -package conduit -predicates mirage_unix key_gen.cmx unikernel.cmx main.cmx -o main.native
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__anon$2dfn$5blib$2ftuntap$2eml$3a24$2c0$2d$2d67$5d_610':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:24: undefined reference to `set_up_and_running'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__anon$2dfn$5blib$2ftuntap$2eml$3a26$2c0$2d$2d52$5d_605':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:26: undefined reference to `get_ifnamsiz'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__open__inner_43':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:34: undefined reference to `tun_opendev'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__closetun_104':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:34: undefined reference to `tun_opendev'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__closetap_126':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:34: undefined reference to `tun_opendev'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__set_ipv4_inner_148':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:46: undefined reference to `set_ipv4'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__get_macaddr_198':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:48: undefined reference to `get_macaddr'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__getifaddrs_386':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:92: undefined reference to `getifaddrs_stub'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__loop_399':
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:96: undefined reference to `iface_get'
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:100: undefined reference to `iface_next'
/home/user/.opam/4.04.0+flambda/build/tuntap.1.4.0/_build/lib/tuntap.ml:101: undefined reference to `freeifaddrs_stub'
/home/user/.opam/4.04.0+flambda/lib/tuntap/tuntap.a(tuntap.o): In function `camlTuntap__string_90':
:(.data+0x368): undefined reference to `iface_next'
:(.data+0x370): undefined reference to `iface_get'
:(.data+0x378): undefined reference to `freeifaddrs_stub'
:(.data+0x380): undefined reference to `getifaddrs_stub'
:(.data+0x388): undefined reference to `get_ifnamsiz'
:(.data+0x390): undefined reference to `set_up_and_running'
:(.data+0x398): undefined reference to `set_ipv4'
:(.data+0x3a0): undefined reference to `get_macaddr'
:(.data+0x3a8): undefined reference to `tun_opendev'
collect2: error: ld returned 1 exit status
```

See also https://travis-ci.org/mirage/mirage-skeleton/jobs/202596593#L1608 .
```

**Pull request merged**